src/launchpad/kafka.py

```diff
@@ -29,234 +29,92 @@
+                logger.error(f"Failed to decode message: {e}")
+                raise  # Re-raise the exception to prevent processing invalid messages
+
+        return RunTaskInThreads(
+            processing_function=process_message,
+            concurrency=self.concurrency,
+            max_pending_futures=self.max_pending_futures,
+            next_step=next_step,
+        )
+
+
+def create_kafka_consumer(
+    message_handler: Callable[[PreprodArtifactEvents], Any],
+) -> StreamProcessor[KafkaPayload]:
+    """Create and configure a Kafka consumer using environment variables."""
+    # Get configuration from environment
+    config = get_kafka_config()
+
+    # Create Arroyo consumer
+    # TODO: When we're closer to production, we'll need a way to disable this logic as
+    # topics, partitions and kafka clusters are configured through getsentry/ops.
+    # We will work with the streaming teams to get this set up.
+    consumer_config = {
+        "bootstrap.servers": config["bootstrap_servers"],
+        "group.id": config["group_id"],
+        "auto.offset.reset": "latest",
+        "enable.auto.commit": False,
+        "enable.auto.offset.store": False,
+    }
 
-    def shutdown(self) -> None:
-        """Signal shutdown to the Kafka consumer."""
-        logger.info("Shutdown requested for Kafka consumer")
-        self._shutdown_requested = True
+    arroyo_consumer = ArroyoKafkaConsumer(consumer_config)
 
-        if self._processor:
-            try:
-                logger.info("Calling signal_shutdown on processor")
-                self._processor.signal_shutdown()
-            except Exception as e:
-                logger.warning(f"Error signaling processor shutdown: {e}")
+    # Create strategy factory
+    strategy_factory = LaunchpadStrategyFactory(
+        message_handler=message_handler,
+        concurrency=config["concurrency"],
+        max_pending_futures=config["max_pending_futures"],
+        healthcheck_file=config.get("healthcheck_file"),
+    )
 
-    def set_message_handler(self, handler: Callable[[LaunchpadMessage], Any]) -> None:
-        """Set the message handler function."""
-        self.message_handler = handler
-        # If processor is already created, we'd need to recreate it
-        # For now, this should be called before start()
-        if self._shutdown_requested:
-            logger.warning("Cannot change message handler while consumer is running")
-
-    async def health_check(self) -> Dict[str, Any]:
-        """Check the health of the Kafka connection."""
-        return {
-            "status": "running" if not self._shutdown_requested else "stopped",
-            "topics": self.topics,
-            "group_id": self.group_id,
-            "bootstrap_servers": self.bootstrap_servers,
-            "processor_active": self._processor is not None,
-        }
+    # Create and return stream processor
+    topics = [Topic(topic) for topic in config["topics"]]
```

Comment:
New to this area but what's the purpose of throwing away all the topics other than `topics[0]`?

Reply:
I think this is related to arroyo's design philosphy, specifically for the StreamProcessor. It's meant to only process messages from one topic with a specific processing strategy.

I think
```
    topics = [Topic(topic) for topic in config["topics"]]
```
should only return an array of size 1 just for our topic, since `config["topics"]` will equal to `preprod-artifact-events`, our topic name